components/tag/selectableList/src/index.js

```diff
@@ -0,0 +1,109 @@
+import React, {Component, PropTypes} from 'react'
+import TagSelectable from '@schibstedspain/sui-tag-selectable'
+
+const removeFromArray = (arr, value) => {
+  return arr.filter(item => value !== item)
+}
+
+export default class TagSelectableList extends Component {
+  state = {
+    selectedValues: []
+  }
+
+  constructor (...props) {
+    super(...props)
+    this.tagHandlers = this.props.tagsList.map((tag) => this.toggleValue.bind(this, tag.value))
+    this.allHandler = this.toggleAll.bind(this)
+  }
+
+  toggleValue (value) {
+    let selectedValues = this.state.selectedValues
+    if (selectedValues.indexOf(value) !== -1) {
+      selectedValues = removeFromArray(selectedValues, value)
+    } else {
+      selectedValues.push(value)
+    }
+    this.setState({selectedValues})
+  }
+
+  toggleAll () {
+    let selectedValues
+    if (this.isAllSelected()) {
+      selectedValues = []
+    } else {
+      selectedValues = this.props.tagsList.map(item => item.value)
+    }
+    this.setState({selectedValues})
+  }
+
+  _renderTags () {
```

Comment:
What about moving the map logic to the render function and here only taking care of rendering the tag?

something like:

```
const _renderTag = (tag, index) => (
 <TagSlectable
  key
  isSelected={this.state.selectedValues.contains(tag.value)}
  ...
}
```

 and in render

```
const {all, tagList} = this.props

return ...
 ....
 tagList.map(this._renderTag)
```